common/persistence/client/store.go

```diff
@@ -90,7 +90,8 @@ func DataStoreFactoryProvider(
 
 	var faultInjection *FaultInjectionDataStoreFactory
 	if defaultCfg.FaultInjection != nil {
-		dataStoreFactory = NewFaultInjectionDatastoreFactory(defaultCfg.FaultInjection, dataStoreFactory)
+		faultInjection = NewFaultInjectionDatastoreFactory(defaultCfg.FaultInjection, dataStoreFactory)
```

Comment:
Yes, previously `faultInjection` was always `nil` in functional and integration tests. Fault injection could still be used if it was specified when setting up the suite (because those params would get passed into `dataStoreFactory`), but it could not be adjusted after the tests had started.